src/pkg/services/m365/users.go

```diff
@@ -66,16 +67,20 @@ func UserHasMailbox(ctx context.Context, acct account.Account, userID string) (b
 		return false, clues.Stack(err).WithClues(ctx)
 	}
 
-	_, err = ac.Users().GetMailInbox(ctx, userID)
-	if err != nil {
-		if err := api.EvaluateMailboxError(err); err != nil {
-			return false, clues.Stack(err)
-		}
+	return exchange.IsServiceEnabled(ctx, ac.Users(), userID)
+}
 
-		return false, nil
+func UserGetMailboxInfo(
```

Comment:
Based on your PR in ark, it looks like with this change i need to do two calls to get this info for the user. I.e

```
if (userHasMailbox) {
   // check mailbox purpose
   }
```

otherwise the errors are not caught when the mailbox doesn't exist. I believe its all a call to the same api, which feels wasteful. Could GetMailbox info catch the errors and return a nil mailbox + a corso error such as "MailboxDoesntExist" ?

Reply:
That sounds like a reasonable request.

Reply:
> Could GetMailbox info catch the errors and return a nil mailbox + a corso error such as "MailboxDoesntExist" ?

Thanks. We do [catch this error](https://github.com/alcionai/corso/blob/main/src/internal/m365/service/exchange/enabled.go#L67) and return an embed MailboxInfo error `api.ErrMailBoxSettingsNotFound`, along with `error = nil`. 

Ark can inspect the embed error to figure out if a user has a mailbox, and then look at the purpose. This way we can eliminate the `UserHasMailbox` call and just rely on `UserGetMailboxInfo`.

Since `api.ErrMailBoxSettingsNotFound` is slightly misleading, I'll also fix it to say `api.ErrMailboxNotFound`.

Also, I am against returning a nil mailbox + corso error here, since it breaks the error reporting contract we have throughout the SDK layer.

Reply:
@juleslasarte @ryanfkeepers  this is ready for a final review

Changes:
1. I've renamed `api.ErrMailBoxSettingsNotFound` to `api.ErrMailboxNotFound`.
2. Introduced `IsErrMailboxNotFound` which can be used by ark (via its m365 connector) to inspect secondary `MailboxInfo` errors & check if a resource has a mailbox. 
3. The corresponding ark PR is now updated with this logic. 

There isn't a better way to do this without disturbing a lot of code, which I feel is an overkill. Also, even if we return `nil, corso error` instead of secondary errors, we would still need ark components to call into the m365 connector to interpret the corso error. Otherwise we risk bleeding corso dependencies into ark components.

Reply:
> we would still need ark components to call into the m365 connector to interpret the corso error

We have a [package for that](https://github.com/alcionai/corso/blob/main/src/pkg/errs/errs.go), actually.  The idea there is that any granularity of low-level error in corso can be compred as a more generic, easily scrutinized sentinel.  That way, SDK users don't go digging into low level packages like m365.  They just check the error to see if it's a, say, "ServiceNotEnabled" err.

Reply:
TIL, thanks. I'll keep this code as is to avoid further refactoring of shared code. It's an extra redirection or two to inspect secondary error code, which is fine imo.